pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,391 @@
+			Fn:           botanist.WaitForExtensionsOperationMigrateToSucceed,
+			Dependencies: flow.NewTaskIDs(annotateExtensionCRsForMigration),
+		})
+		deleteAllExtensionCRs = g.Add(flow.Task{
+			Name:         "Deletes all extension CRs from the Shoot namespace",
+			Fn:           botanist.DeleteAllExtensionCRs,
+			Dependencies: flow.NewTaskIDs(waitForExtensionCRsOperationMigrateToSucceed),
+		})
+		waitUntilAPIServerScaledDown = g.Add(flow.Task{
+			Name:         "Wait for Kubernetes API Server to be scaled to zero",
+			Fn:           flow.TaskFn(botanist.WaitUntilKubeAPIServerScaledDown).SkipIf(shootHibernated || !nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeploymentsForMigration),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Wait for gardener-resource-manager to be scaled to zero",
+			Fn:           flow.TaskFn(botanist.WaitUntilGardenerResourceManagerScalesDown).SkipIf(shootHibernated || !nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeploymentsForMigration),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying ingress DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyIngressDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerScaledDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying external domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyExternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerScaledDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying internal domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerScaledDown),
+		})
+		createETCDSnapshot = g.Add(flow.Task{
+			Name:         "Create ETCD Snapshot",
+			Fn:           flow.TaskFn(botanist.CreateETCDSnapshot).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerScaledDown),
+		})
+		scaleETCDToZero = g.Add(flow.Task{
+			Name:         "Scale ETCD to zero",
+			Fn:           flow.TaskFn(botanist.ScaleETCDToZero).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(createETCDSnapshot),
+		})
+		deleteNamespace = g.Add(flow.Task{
+			Name:         "Deleting shoot namespace in Seed",
+			Fn:           flow.TaskFn(botanist.DeleteNamespace).Retry(defaultInterval),
+			Dependencies: flow.NewTaskIDs(deleteAllExtensionCRs, waitManagedResourcesDeletion, scaleETCDToZero),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Waiting until shoot namespace in Seed has been deleted",
+			Fn:           botanist.WaitUntilSeedNamespaceDeleted,
+			Dependencies: flow.NewTaskIDs(deleteNamespace),
+		})
+
+		f = g.Compile()
+	)
+
+	if err := f.Run(flow.Opts{Logger: o.Logger, ProgressReporter: o.ReportShootProgress, ErrorCleaner: o.CleanShootTaskError}); err != nil {
```

Comment:
Sorry for noticing so late. Can you pass the errorContext here as well:
```suggestion
	if err := f.Run(flow.Opts{Logger: o.Logger, ProgressReporter: o.ReportShootProgress, ErrorContext: errorContext, ErrorCleaner: o.CleanShootTaskError}); err != nil {
```